### PR TITLE
Redo A&S 100 and 120 Credit Minimum Requirements

### PR DESCRIPTION
**Pull request by noschiff**
### Summary 

- [x] Use existing 120 Total Credit requirement for A&S.
- [x] Excludes transfer (including AP) credits from counting for the required 100 credits from Arts and Sciences.

The [A&S Degree Requirements page](**url**) links to a [**website with information**](https://courses.cornell.edu/content.php?catoid=45&navoid=17977#credit-req) about the credit requirement, including exclusions for 100 & 120 requirements:

> The College of Arts and Sciences does not grant credit toward the degree for every course offered by the university. Courses in military training, service as a teaching assistant, physical education, remedial or developmental training, precalculus mathematics, supplemental science and mathematics, offered by the Learning Strategies Center and English as a second language are among those for which degree credit.

> Courses that do not count toward the 100 required Arts and Sciences credits include credits earned in other colleges at Cornell (except in the cases specifically noted in this section), transfer credits earned in any subject at institutions other than Cornell, and advanced placement/test credits. **AP/test credits** count as part of the 120 credits required for the degree but **not as part of the 100 Arts and Sciences credits** and may not be applied to distribution requirements. AP credits are posted on the transcript. If, subsequently, a student takes the course out of which they had placed, the AP credit will be removed because of the overlap in content.

### Test Plan 

1. Check that all credits (except for courses numbered 1000 - 1099 and PE classes) count for the 120 Total requirements. _Note that cross-listed PE courses will incorrectly count for the 120 total; this needs to be addressed later but it will require a bigger change than we have time for now._
2. Ensure that AP credits are not counted towards the 100 A&S credits. ~~Note that this PR was made **before** #684 was merged into master, so **AP credits are disabled by default for every requirement.**~~ _This should not be an issue anymore since master was updated, but it would be helpful to still_ run `GK.enableAPIBFulfillment()` in the console first, and it would be helpful to make sure that your AP credits still count for something they should (like AP Computer Science A for Computer Science major in A&S).
3. Ensure that only classes that count for the 120 Total Credits can count for the 100 A&S Credits. The 100 A&S Credits **must come from the 120 Total Credits**.

### Notes

- I don't currently include all the courses from the [Courses That Don't Count](https://as.cornell.edu/registrar/courses-that-dont-count) website.
- I'm not sure about the wording for the descriptions.
- I'd like to link to the [Courses That Don't Count](https://as.cornell.edu/registrar/courses-that-dont-count) website, but our source ("Learn More" link) usually goes to a more general page, not a page of exclusions.
- I only added the 120 total credit requirement to the current A&S college. Do I also add it to pre 2020?

### CODE REVIEW

**Comment by noschiff** [comment]
is this better suited by compound requirement? i think it's helpful for the user to see both the 100 and 120 but they do share a lot of courses

File: src/requirements/data/checkers-common.ts
Comment (willespencer) [inline src/requirements/data/checkers-common.ts:107]: This is good! Putting in all the disallowed courses would be difficult, and if a course ever became re-allowed, we would be preventing users from using valid courses

File: src/requirements/data/colleges/asFA2020.ts
Comment (willespencer) [inline src/requirements/data/colleges/asFA2020.ts:11]: We already have a 120 total credit check in the pre-FA20 A&S college. Not sure why it never got implemented here, but could you model this one after that one to be consistent?

File: src/requirements/data/colleges/asFA2020.ts
Comment (willespencer) [inline src/requirements/data/colleges/asFA2020.ts:21]: There is code somewhere that makes the pre-FA2020 check to not show all of the courses that fulfill it - because the list would be very long and it is essentially just a long list of everything in your semester plan. Could you do the same here?

image

File: src/requirements/data/checkers-common.ts
Comment (willespencer) [inline src/requirements/data/checkers-common.ts:105]: This req name implies it will only be used for the A&S 100 credit req, but it is used for the 120 credit req too. Is there a more clear name you could use?

**Review by willespencer — CHANGES_REQUESTED** [review]
Left some comment suggestions - mostly about making sure this better matches our pre FA20 A&S req group!

File: src/requirements/__test__/__snapshots__/requirement-data-id.test.ts.snap
Comment (noschiff) [reply src/requirements/__test__/__snapshots__/requirement-data-id.test.ts.snap:22]: there shouldn't be? I'm confused… what changed? if you're talking about the red highlight on the `&`, I'm pretty sure that github just does that for some reason

File: src/requirements/data/checkers-common.ts
Comment (noschiff) [reply src/requirements/data/checkers-common.ts:107]: thanks!

File: src/requirements/data/colleges/asFA2020.ts
Comment (noschiff) [reply src/requirements/data/colleges/asFA2020.ts]: yup, just figured that out

File: src/requirements/data/colleges/asFA2020.ts
Comment (noschiff) [reply src/requirements/data/colleges/asFA2020.ts]: deleted this and copied the pre2020 one

Comment (noschiff) [reply src/requirements/data/colleges/asFA2020.ts]: it's not as accurate because it circumvents our usual way of checking requirements, but it'll do for now

File: src/requirements/data/colleges/asFA2020.ts
Comment (noschiff) [reply src/requirements/data/colleges/asFA2020.ts]: ok! the link has to be different because there's a different page for pre 2020

**Review by willespencer — APPROVED** [review]
Code looks much better! Let's wait for approval from @einc as discussed offline, but then it should be good to merge!

**Pull request merged**
